39/mcjsk/netcalc.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+"""
+netcalc IPv4 subnet calculator
+
+Return range of contiguous block of addresses determined by the
+CIDR notation specified on the command line. See --help for usage.
+
+Logic:
+  1. Split command line specified CIDR into list of IPv4 address and prefix
+  2. Type cast prefix and octets in list to integer for arithmetic operation
+  3. Compute subnet block length from host bits (i.e., the remainder after
+  subtracting mask bits (prefix) from the 32-bit address space
+  4. Compute prefix as a 32-bit integer representation of the subnet mask
+  5. Convert list of octets to 32-bit integer representation of IP address,
+  then logical AND with subnet mask to compute start IP address of subnet
+  6. Compute 32-bit integer representation of end IP address of subnet by
+  adding subnet block length to start IP address integer
+  7. Convert start, end, and bitmask IP integers to string representation
+  of IPv4 address in dot notation
+"""
+__author__ = "Mark Jamsek"
+__version__ = "0.3"
+__date__ = "18 July 2019"
+__email__ = "mark@jamsek.com"
+
+import argparse
+
+
+def main():
+    """
+    Verify command line arguments are valid:
+      - correct CIDR (i.e., dash "/") notation
+      - CIDR prefix is an integer
+      - IPv4 address is correct length
+      - IPv4 address is numeric
+    If invalid, print reason for error and usage example upon exit.
+    If valid, proceed with program logic.
+    """
+    args, parser = parse_commands()
+    cidr = args.cidr[0].split("/")
+    if len(cidr) != 2:
+        print(f"\nError: incorrect notation {args.cidr}")
+        exit(f"{parser.format_usage()}")
+    ipstring = cidr[0]
+    try:
+        prefix = int(cidr[1])
+        blocklen = 2 ** (32 - prefix)
+    except:
+        print(f"\nError: CIDR prefix must be integer ['{cidr[1]}']")
+        exit(f"{parser.format_usage()}")
+    iplist = ipstring.split(".")
+    if len(iplist) != 4:
+        print(f"\nError: abnormal length IP address ['{ipstring}']")
+        exit(f"{parser.format_usage()}")
+    try:
+        iplist = [int(x) for x in iplist]
+    except:
+        print(f"\nError: illegal characters in IP address ['{ipstring}']")
+        exit(f"{parser.format_usage()}")
+
+    bitmask_int = mask_to_int(prefix)
```

Comment:
this function is pretty long, I would split it into multiple functions and separate "calculation" from printing logic